### ISSUE TITLE
Build request object within dispatcher if not already defined in container

### ISSUE THREAD

**Issue by ry167** [thephpleague/route#26]
In `RequestResponseStrategy.php`, when `Symfony\Component\HttpFoundation\Request` is fetched from the League Container, if it is not already set in the container then it will initiate a largely useless Request object that contains none of the globals that HttpFoundation Request is normally initiated with.

I am not extremely familiar with Dependency Injection so I don't know what the ideal fix for this is in this package. When using Route in a project you can make a Service Provider for HttpFoundation or set it with `$container->add('Symfony\Component\HttpFoundation\Request', Request::createFromGlobals());`, but it was quite unexpected to have to do so.

**Comment by philipobenito** [comment]
This is something I've been thinking about recently and I think you're right, we should probably build the request object within the dispatcher if it has not already been defined.

**Comment by hannesvdvreken** [comment]
@ry167 I would wrap that in a closure:

```php
$container->add('Symfony\Component\HttpFoundation\Request', function () {
    return Request::createFromGlobals();
});
```

**Comment by philipobenito** [comment]
Apologies for the delay in addressing this, for v1 I will be implementing a check for the whether the request has been instantiated properly or not and then building it if not. v2 wll handle it slightly differently as there is a change around the HTTP layer of dependencies.

**Comment by philipobenito** [comment]
Fixed by 803cad311c52d91e5b7fef5e7abcde4ea52409d5
